homeassistant/components/abode/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for the Abode Security System component."""
+import logging
+import voluptuous as vol
+from abodepy import Abode
+from abodepy.exceptions import AbodeAuthenticationException
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AbodeFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Abode."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self.data_schema = {}
```

Comment:
We could make this a `Schema` instance directly.

Reply:
I may have messed this one up and will need to fix. I updated to: 
```
        self.data_schema = vol.Schema(
            {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
        )
```
Thought it worked when I tried it but it's throwing `TypeError: unhashable type: 'Schema'` now.

Reply:
Fixed. Will be in the next commit:
```
        self.data_schema = {
            vol.Required(CONF_USERNAME): str,
            vol.Required(CONF_PASSWORD): str,
        }
```